api/client/proxy/transport/transportv1/client.go

```diff
@@ -56,6 +60,98 @@ func (c *Client) ClusterDetails(ctx context.Context) (*transportv1pb.ClusterDeta
 	return resp.Details, nil
 }
 
+const (
+	// DesktopSNISuffix is the server name suffix used during SNI to specify the
+	// target desktop to connect to. The client will use SNI
+	// like "${UUID}.desktop.teleport.cluster.local" to pass the UUID of the
+	// desktop.
+	// This is a copy of the same constant in `lib/srv/desktop/desktop.go` to
+	// prevent depending on `lib/srv` in `api/client`.
+	DesktopSNISuffix = ".desktop." + constants.APIDomain
+)
+
+// ProxyWindowsDesktopSession establishes a connection to the target desktop over a bidirectional stream.
+// The caller is required to pass a valid desktop certificate.
+func (c *Client) ProxyWindowsDesktopSession(ctx context.Context, cluster string, desktopName string, desktopCert tls.Certificate, rootCAs *x509.CertPool) (net.Conn, error) {
+	connCtx, cancel := context.WithCancel(context.WithoutCancel(ctx))
```

Comment:
I wonder if we should add something like 
```go
success := false
defer func(){
  if !success {
    cancel()
  }
}()
```
and set `success = true` before the last `return`. That way we won't forget to cancel context if we add more code later

Reply:
I decided for the dial method, I think it's slightly more explicit.